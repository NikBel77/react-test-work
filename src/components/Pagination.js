import React from 'react'
import leftArrow from '../icons/left-arr.svg'
import rightArrow from '../icons/right-arr.svg'
import { countPaginationNumbers } from '../utils'

function Pagination(props) {
    const pagination = countPaginationNumbers(props.currentPage, props.numberOfPages);
    const leftArrowClass = pagination[0] === props.currentPage ? 'page-item disabled' : 'page-item';
    const rightArrowClass = pagination[pagination.length - 1] === props.currentPage ? 'page-item disabled ' : 'page-item';

    return (
        <div>
            <div className='pagination-wrapper'>
                <ul className='pagination'>
                    <li className={leftArrowClass}>
                        <img className='page-link' src={leftArrow} alt='arrow'/>
                    </li>

                    {pagination.map((el, i) => {
                        if (el === props.currentPage) {
                            return (
                                <li key={i} className='page-item active'>
                                    <span className='page-link'>{el}</span>
                                </li>
                            )
                        } else {
                            return (
                                <li key={i}
                                    className='page-item'
                                    onClick={() => props.onPageChange(el)}>
                                    <span className='page-link'>{el}</span>
                                </li>
                            )
                        }
                    })}

                    <li className={rightArrowClass}>
                        <img className='page-link' src={rightArrow} alt='arrow'/>
                    </li>
                </ul>
            </div>
        </div>
    )
}

export default Pagination