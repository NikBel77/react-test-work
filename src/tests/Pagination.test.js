import React from 'react'
import ReactDOM from 'react-dom'
import Pagination from '../components/Pagination'
import renderer from 'react-test-renderer'
import { render } from '@testing-library/react'

test('render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Pagination
        numberOfPages={10}
        currentPage={5}
    />, div);
    ReactDOM.unmountComponentAtNode(div);
});
test('no render with 1 page', () => {
    const { queryByText } = render(<Pagination
        numberOfPages={1}
        currentPage={1}
    />);
    expect(queryByText(/^\d+$/)).toBeNull();
});
test('snapshot', () => {
    const tree = renderer.create(<Pagination
        numberOfPages={10}
        currentPage={5}
    />).toJSON();
    expect(tree).toMatchSnapshot();
});