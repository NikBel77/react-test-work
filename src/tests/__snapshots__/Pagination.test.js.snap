// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div>
  <div
    className="pagination-wrapper"
  >
    <ul
      className="pagination"
    >
      <li
        className="page-item"
        onClick={[Function]}
      >
        <img
          alt="arrow"
          className="page-link"
          src="left-arr.svg"
        />
      </li>
      <li
        className="page-item"
        onClick={[Function]}
      >
        <span
          className="page-link"
        >
          2
        </span>
      </li>
      <li
        className="page-item"
        onClick={[Function]}
      >
        <span
          className="page-link"
        >
          3
        </span>
      </li>
      <li
        className="page-item"
        onClick={[Function]}
      >
        <span
          className="page-link"
        >
          4
        </span>
      </li>
      <li
        className="page-item active"
      >
        <span
          className="page-link"
        >
          5
        </span>
      </li>
      <li
        className="page-item"
        onClick={[Function]}
      >
        <span
          className="page-link"
        >
          6
        </span>
      </li>
      <li
        className="page-item"
        onClick={[Function]}
      >
        <span
          className="page-link"
        >
          7
        </span>
      </li>
      <li
        className="page-item"
        onClick={[Function]}
      >
        <span
          className="page-link"
        >
          8
        </span>
      </li>
      <li
        className="page-item"
        onClick={[Function]}
      >
        <img
          alt="arrow"
          className="page-link"
          src="right-arr.svg"
        />
      </li>
    </ul>
  </div>
</div>
`;
